mshadow/half2.h

```diff
@@ -85,31 +85,31 @@ class half2_t {
 /*! \brief overloaded + operator for half2_t */
 MSHADOW_XINLINE half2_t operator+(half2_t a, half2_t b) {
 #if MSHADOW_CUDA_HALF2
-  return half2_t(__hadd2(a.half2_, b.half2_));
+  return half2_t(__halves2half2(__low2float(a.half2_) + __low2float(b.half2_), __high2float(a.half2_) + __high2float(b.half2_)));
 #else
   return half2_t(a.half_t2[0] + b.half_t2[0], a.half_t2[1] + b.half_t2[1]);
 #endif
 }
 /*! \brief overloaded - operator for half2_t */
 MSHADOW_XINLINE half2_t operator-(half2_t a, half2_t b) {
 #if MSHADOW_CUDA_HALF2
-  return half2_t(__hsub2(a.half2_, b.half2_));
+  return half2_t(__halves2half2(__low2float(a.half2_) - __low2float(b.half2_), __high2float(a.half2_) - __high2float(b.half2_)));
```

Comment:
Few reasons - 
 - using intrinsics forces usage of fp16 even on gpus that do not have fast fp16 math (like Titan XP or 1080ti) and this may be a performance problem
 - for memory bound kernels (and those operations will mostly be used in pointwise kernels that are memory bound it does not matter for GPUs with fast fp16 math
 - precision is better that way (also, it is consistent with what you do in half.h for normal halfs)